src/d3a/models/market/one_sided.py

```diff
@@ -245,11 +247,13 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
 
         # Delete the accepted offer from self.offers:
         self.offers.pop(offer.id, None)
-
+        fee_price = fee_rate * offer.energy
```

Comment:
I think that `self.market_fee` should be updated here, in order to cover all market types. Needs to also take into account the `already_tracked` flag. Also please clone the changes to the 2 sided market too, thanks!